### DISCUSSION TITLE
Pause/Resume librespot headless playback

### DISCUSSION THREAD

**Discussion by DJ1TJOO** [librespot-org/librespot#1413] [General]
Hello,

I have librespot installed on a windows machine using cargo. It is running in headless mode with discovery on. So my house mates and I can connect to it. Please note that I'm not experienced in anything regarding librespot except for installing it :).

I would like to know if it is possible to pause/resume librespot using external tools, either by a command or interacting with it in some way. I do not currently know any rust, but I'm willing to learn!

I did see a thread of something similair but it was two years old, so I thought I would ask again. https://github.com/librespot-org/librespot/issues/457. 
https://github.com/librespot-org/librespot/pull/1341 seems also usefull but not impletemented yet, should I wait for this?

I also looked at spotifyd, but I thought it was not possible to run it on windows?

Any help, suggestions, or links to other threads would be appreciated!

**Comment by photovoltex** [comment]
Currently there isn't any option provided by the librespot binary (the executable you installed using `cargo install`). The library itself does expose methods to manipulate the playback of librespot itself, but that isn't used by the binary, yet (as you saw it may come in the future).

So far as I know when using discovery you need to use an official client to initially connect to librespot, after that you can use any client that can control spotify-connect devices. Such as the official clients or a third party client like [spotify-tui](https://github.com/Rigellute/spotify-tui). Third party clients usually just use the [web-api](https://developer.spotify.com/documentation/web-api), so maybe that's already enough for your need (even tho that usually requires more fiddling).

**Reply by photovoltex** [reply]
Also related: #1119, #1050, #756
